crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1039,6 +1056,28 @@ impl<'w> EntityWorldMut<'w> {
         }
     }
 
+    /// Consumes `self` and returns read-only access to all of the entity's
+    /// components, with the world `'w` lifetime.
+    pub fn into_readonly(self) -> EntityRef<'w> {
+        EntityRef::from(self)
+    }
+
+    /// Gets read-only access to all of the entity's components.
+    pub fn as_readonly(&self) -> EntityRef<'_> {
+        EntityRef::from(self)
+    }
+
+    /// Consumes `self` and returns non-structural mutable access to all of the
```

Comment:
What does "non-structural" mean in this context?

Reply:
Structural access (`&mut World` access) means you can insert/remove components, or despawn the entity. Non-structural access allows none of that; instead you have sliced access to the world which means you aren't allowed to perform changes that might affect other entities.

Reply:
I see, thanks for the clarification!